recipes/generate_keytabs.rb

```diff
@@ -64,6 +64,7 @@
       command "kadmin -w #{node['krb5_utils']['admin_password']} -q 'xst -kt #{keytab_dir}/#{keytab_file} #{principal} #{http_principal}'"
       action :run
       not_if "test -e #{keytab_dir}/#{keytab_file}"
+      creates "#{keytab_dir}/#{keytab_file}"
```

Comment:
Not really. A keytab can only be generated once. If you ever regenerate it, the KDC invalidates all others. It should be done only once. If you ever had a reason to regenerate them, simply deleting the old ones would suffice to have them regenerated. I don't think that we need to directly support that use case in the cookbook.